readthedocs/notifications/decorators.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+
+import logging
+
+from django.db.models import Q
+
+from readthedocs.projects.models import Project
+from readthedocs.projects.notifications import DeprecatedWebhookEndpointNotification
+
+log = logging.getLogger(__name__)
+
+
+def notify_deprecated_endpoint(function):
+    """
+    Decorator to notify owners that the endpoint is DEPRECATED.
+    """
+    def wrap(request, *args, project_id_or_slug=None, **kwargs):
+        try:
+            project = Project.objects.get(
+                Q(pk=project_id_or_slug) | Q(slug=project_id_or_slug),
+            )
+        except (Project.DoesNotExist, ValueError):
+            log.info('Project not found: slug=%s', project_id_or_slug)
+
+        # Send one notification to each owner of the project
+        for user in project.users.all():
```

Comment:
Is there a better query that accommodates logic from the commercial code? I'm not sure if we need this or not on the commercial side, I'm mostly certain that no one is hitting the `/github` endpoint. Worth exploring though.

Reply:
I checked the logs and I didn't find anyone using these endpoints in the commercial hosting: https://github.com/rtfd/readthedocs-corporate/issues/363#issuecomment-438648073